dashboard/app/models/pd/application/facilitator_application_base.rb

```diff
@@ -413,7 +415,29 @@ def application_url
 
     # override
     def enroll_user
-      super
+      return unless pd_workshop_id
+
+      enrollment = Pd::Enrollment.where(
+        pd_workshop_id: pd_workshop_id,
+        email: user.email
+      ).first_or_initialize
+
+      # If this is a new enrollment, we want to:
+      #   - save it with all required data
+      #   - save a reference to it in properties
+      #   - delete the previous auto-created enrollment if it exists
+      if enrollment.new_record?
+        enrollment.update(
+          user: user,
+          school_info: user.school_info,
+          full_name: user.name
+        )
+        enrollment.save!
```

Comment:
I know a lot of these are just copied, just calling things out as I go.

Doesn't `update` implicitly save?  I think if we use `update!` here we can remove the `save!` line entirely.